### DISCUSSION TITLE
Email change workflow

### DISCUSSION THREAD

**Discussion by DaBonse** [keycloak/keycloak#12235] [Q&A]
We use email address as user name in Keycloak. This cause one problem:

### This is the email change workflow of Keycloak 15.0.2, we use:

1.) User changes his email address
2.) No verification email was sent
3.) User have to log off, and log on
4.) Verification email was sent
5.) User can verify his email address
6.) User can login in with new email address

### Problem:

If the user mistype his new email address, no verification email can be received. And the old email address ist not longer working for login. User can not log in anymore. 

Our user do not know an username, because we do not use user names for login.

In this case, the workflow should look like this:

1.) User changes his email address
2.) A verification email was sent
3.) If the user do not verify the new email, the old email still works for login
4.) User verify email address, the old email address will be replaced with the new one
5.) User can login with new email address. The old email address no longer works for login

### Question:

Is there a way in Keycloak to change this workflow to the one I suggest?

Is there a email history in Keycloak, so we can manual check the old email address?

Thanks,
DaBonse

**Comment by akikoskinen** [comment]
This could be related https://github.com/keycloak/keycloak/pull/7943/

**Comment by DaBonse** [comment]
We use now the AIA UPDATE_EMAIL Feature, that fits our needs.

**Comment by vi-buuuk** [comment]
Hi @DaBonse 

Current I follow instruction flow of keycloak with update_email action, but when I go to account console to update email, it just return account console after I submit new email. It doesn't send verification.

Can I ask that there is something changes in configurations for missing step verify email ?
